app/models/course.rb

```diff
@@ -1,4 +1,9 @@
 class Course < ApplicationRecord
+
+  # associations
+  has_many :user_courses
+  has_many :users, through: :user_courses
+
   enum expertise: { beginner: 0, intermediate: 1, expert: 2 }
   # validations
   validates_presence_of :title, :expertise, :created_by
```

Comment:
Should have linked to an author with
belongs_to :author

Reply:
I have some reservations about having an author model. We have a User model in which an Author can also be a User. A student is also a type of User. I'm thinking this should be a polymorphic association. What do you think?

Reply:
The point being that author is referenced. you don't need a polymorphic association
belongs_to :author, class_name: 'User' will work and put it on the migration file too

Reply:
Hmmmm, true. Thanks for the tip